.github/workflows/deploy-ghpages.yaml

```diff
@@ -0,0 +1,32 @@
+# Build the project and deploy it to GitHub pages
+name: Deploy
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: [master, main, production]
+
+jobs:
+  deploy:
+    name: Build and Deploy API
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout source repository
+      uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      shell: bash
+      run: ./ci/install.sh
+    - name: Build API content
+      shell: bash
+      run: ./ci/build.sh
+    - name: Deploy to GitHub Pages
+      uses: JamesIves/github-pages-deploy-action@v4
+      with:
+        BRANCH: gh-pages    # The branch the action should deploy to
+        FOLDER: ./fsf-api-build-output    # The folder the action should deploy
+        CLEAN: true    # Automatically remove deleted files from the deploy branch
```

Comment:
Turns out that there is a file not generated - [index.html](https://github.com/spdx/fsf-api/blob/gh-pages/index.html).

Perhaps we should keep the index.html file in the master branch and have this action copy the file to the fsf-api-build-output folder.  That would be cleaner than the current approach where the file is maintained in the gh-pages branch.

Reply:
Great catch, and I definitely agree with the approach. The only question is whether including the index file is considered a part of the API and should be part of generating the API data, and thus be copied to the output dir by the pull.py script, or should be considered a property of the deployment to GitHub pages, and thus copied to the output dir in the CI-specific build script. After giving some thought to the matter, and initially leaning toward the former, I'm fairly convinced of the latter, since its specific format and contents are specific to this particular deployment on a static site and would not be applicable to most other use cases, and also matches the current workflow.

I'll stick the `index.html` (as well as a `.nojekyll` to speed up deployment and avoid any wierd behavior) in an `assets` directory and `cp` it in the `build` script. I'll also move the check script to an actual Python script in the `ci` directory, for modularity.

Reply:
All done!